### PR TITLE
Fix equals() and hashCode() for AccountReadableKVState

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:
This PR fixes equals() and hashCode(), so that the Supplier fields are correctly compared.

It also adds additional tests for testing account alias population.

**Related issue(s)**: #9251 

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by kselveliev — APPROVED** [review]

File: hedera-mirror-web3/src/main/java/com/hedera/hapi/node/state/token/Account.java [hunk 1]
```diff
@@ -284,7 +286,9 @@ public int hashCode() {
         if (headNftSerialNumber != DEFAULT.headNftSerialNumber) {
             result = 31 * result + Long.hashCode(headNftSerialNumber);
         }
-        if (!numberOwnedNftsSupplier.equals(DEFAULT.numberOwnedNftsSupplier)) {
+        if (numberOwnedNftsSupplier != null
```
Comment (bilyana-gospodinova) [inline hedera-mirror-web3/src/main/java/com/hedera/hapi/node/state/token/Account.java:289]: If one of the suppliers is `null`, the whole `equals` comparison will be skipped. Is this expected?

Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/hapi/node/state/token/Account.java:289]: Good catch! We'll need additional checks to cover all cases.

**Review by bilyana-gospodinova — APPROVED** [review]
LGTM

**Review by kselveliev — APPROVED** [review]

**Pull request merged**